task/bulk_task.go

```diff
@@ -71,11 +73,12 @@ func newBulkTask(
 		listener:          listener,
 		releaseTasksCh:    releaseTasksCh,
 		gasLimitBumpRatio: gasLimitBumpRatio,
+		maxBulkTasks:      maxBulkTasks,
 	}
 }
 
 func (r *bulkTask) collectTask(t *models.Task) {
-	if t.Type == r.taskType {
+	if t.Type == r.taskType && (r.maxBulkTasks == 0 || len(r.tasks) <= r.maxBulkTasks) {
 		r.tasks = append(r.tasks, t)
 	}
```

Comment:
```suggestion
if t.Type == r.taskType && (r.maxBulkTasks == 0 || len(r.tasks) <= r.maxBulkTasks) {
		r.tasks = append(r.tasks, t)
		return
}
log.Info(("[RoninTask][collectTask] Drop task because reaching maxBulkTasks", "threshold", r.maxBulkTasks)
```